fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -1741,6 +1734,15 @@ Partition createPartition(Database db, OlapTable table, long partitionId, String
             MaterializedIndex index = entry.getValue();
             MaterializedIndexMeta indexMeta = table.getIndexIdToMeta().get(indexId);
 
+            // create shard group
+            long shardGroupId = PhysicalPartitionImpl.INVALID_SHARD_GROUP_ID;
+            if (table.isCloudNativeTableOrMaterializedView()) {
+                long indexShardGroupId = getNextId();
+                shardGroupId = GlobalStateMgr.getCurrentState().getStarOSAgent().
+                        createShardGroup(db.getId(), table.getId(), indexShardGroupId);
+                index.setShardGroupId(shardGroupId);
```

Comment:
better to set `partition shardGroupId` same as the base index for old version rollback.